CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
I played a little in compiler explorer, and gcc seems fairly good about consistently using fma where possible (even with `-O0`).

But my concerns about cases like `lerp` remain. We would be loosing the guarantee that:

`lerp(x, a, b) == lerp(1 - x, b, a)`

Which I think some code implicitly assumes.